extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -0,0 +1,661 @@
+'use strict'; // eslint-disable-line
+
+const async = require('async');
+
+const errors = require('arsenal').errors;
+
+const BackbeatTask = require('../../../lib/tasks/BackbeatTask');
+
+const RETRYTIMEOUTS = 300;
+
+// Default max AWS limit is 1000 for both list objects and list object versions
+const MAX_KEYS = 1000;
+
+class LifecycleTask extends BackbeatTask {
+    /**
+     * Process a list of versioned or non-versioned objects
+     *
+     * @constructor
+     * @param {LifecycleProducer} lp - lifecycle producer instance
+     */
+    constructor(lp) {
+        const lpState = lp.getStateVars();
+        super({ retryTimeoutS: RETRYTIMEOUTS });
+        Object.assign(this, lpState);
+    }
+
+    /**
+     * Handles non-versioned objects
+     * @param {object} bucketData - bucket data
+     * @param {object} params - params for AWS S3 `listObjectsV2`
+     * @param {Logger.newRequestLogger} log - logger object
+     * @param {function} done - callback(error, data)
+     * @return {undefined}
+     */
+    _getObjectList(bucketData, params, log, done) {
+        this.s3target.listObjectsV2(params, (err, data) => {
+            if (err) {
+                log.error('error listing bucket objects', {
+                    error: err,
+                    bucket: params.Bucket,
+                });
+                return done(errors.InternalError);
+            }
+            if (data.IsTruncated) {
+                // re-queue to Kafka topic bucketTasksTopic
+                // with bucket name and `data.NextContinuationToken`
+                const entry = Object.assign({}, bucketData, {
+                    continuationToken: data.NextContinuationToken,
+                });
+                this.sendBucketEntry(entry, () => {
+                    log.debug('sending kafka entry for bucket consumption', {
+                        method: 'LifecycleTask._getObjectList',
+                    });
+                });
+            }
+            if (data.Contents.length === 0) {
+                // if bucket is empty, just skip
+                log.debug('no objects in bucket', {
+                    bucket: params.Bucket,
+                });
+                return done();
```

Comment:
Since within `_compareWithLCRules`, `data.Contents` will not be defined, the next step `_compareContents` will be skipped since I just check if the param is defined or not. I know this is a bit confusing and am hoping to refactor all this in the coming weeks, but for now, I think I should not set `data.Contents` to anything